ENH: Use ruff for formatting instead of black + update to 0.5.5 (#3397)
geopandas/geopandas: doc/source/community/contributing.rst

  line of a docstring should be a standalone summary.  Parameters and
  return values should be documented explicitly.

> - Follow PEP 8 when possible. We use `ruff <https://beta.ruff.rs/docs/>`_

Reviewer: ```suggestion
- Follow PEP 8 when possible. We use `ruff <https://docs.astral.sh/ruff/>`_
```
this is where the old one redirects

Style Guide & Linting
---------------------

GeoPandas follows the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_

> standard and uses `ruff <https://beta.ruff.rs/docs/>`_ to ensure a consistent

Reviewer: ```suggestion
standard and uses `ruff <https://docs.astral.sh/ruff/>`_ to ensure a consistent
```

code format throughout the project.

Continuous Integration (GitHub Actions) will run those tools and
report any stylistic errors in your code. Therefore, it is helpful before
submitting code to run the check yourself::

>    ruff geopandas

Reviewer: ```suggestion
   ruff format geopandas
   ruff check geopandas
```
`ruff <path>` is deprecated and if you're replacing black, you want `ruff format` anyway
